File: 프로그래머스/2/42587. 프로세스/프로세스.cs
using System;

public class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 1;
        
        int[] process = (int[])priorities.Clone();
        Array.Sort(priorities);
        Array.Reverse(priorities);
        int index =0;
        int max = priorities[index++];
        
        while(priorities.Length != 0){
            for(int i =0; i<process.Length; i++){
              
                if(process[i] != max)
                  continue;
                
                    if(i==location)
                        return answer;
                  
                    max = priorities[index++];
                    answer++;
                
            }
       }
        
        return answer;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 0;
    priority_queue<int> waitQ;
    
    for(int p : priorities){
        waitQ.push(p);
    }

    while(waitQ.size()>0)
    {
        for(int i= 0; i<priorities.size();i++){
            if(waitQ.top() == priorities[i])
            {
                answer++;
                waitQ.pop();
                
                if(i == location){
                    return answer;
                }
            }
        }
    }    
}